<?xml version="1.0"?>
<!--
使用范例:
-->
<launch>
  <!-- 调试等级 -->
  <arg name="verbose" default="0" />
  <!-- msg.header.frame_id -->
  <arg name="frame_id" default="livox_frame" />
  <!-- 是否发布深度图(基于彩色相机参数) -->
  <arg name="output_depth_on_rgb" default="1" />
  <!-- 是否发布深度图(基于热感相机参数) -->
  <arg name="output_depth_on_thermal" default="1" />
  <!-- 是否融合热感图到彩色图，1-自动激活output_depth_on_rgb -->
  <arg name="rgb_fuse_with_thermal" default="1" />
  <!-- 通过HSV变换，把彩色图伪造为热感图输出(新加) -->
  <arg name="HSV_h1" default="1.0" />
  <arg name="HSV_h2" default="1" />
  <arg name="HSV_s1" default="1.0" />
  <arg name="HSV_s2" default="1" />
  <arg name="HSV_v1" default="1.0" />
  <arg name="HSV_v2" default="1" />
  <group ns="/sensors_calib">
    <node pkg="sensors_calib" type="sensors_calib_node" name="sensors_calib_node" output="screen" >
        <param name="frame_id" value="$(arg frame_id)" />
        <param name="verbose" type="int" value="$(arg verbose)" />
        <param name="output_depth_on_rgb" type="bool" value="$(arg output_depth_on_rgb)" />
        <param name="output_depth_on_thermal" type="bool" value="$(arg output_depth_on_thermal)" />
        <param name="rgb_fuse_with_thermal" type="bool" value="$(arg rgb_fuse_with_thermal)" />
        <param name='intrinsic_rgb' value='$(find sensors_calib)/config/intrinsic_rgb.txt' />
        <param name='intrinsic_thermal' value='$(find sensors_calib)/config/intrinsic_thermal.txt' />
        <param name='extrinsic_livox2rgb' value='$(find sensors_calib)/config/extrinsic_livox2rgb.txt' />
        <param name='extrinsic_livox2thermal' value='$(find sensors_calib)/config/extrinsic_livox2thermal.txt' />
        <param name='extrinsic_thermal2rgb' value='$(find sensors_calib)/config/extrinsic_thermal2rgb.txt' />
        <param name='homograph_thermal2rgb' value='$(find sensors_calib)/config/homograph_thermal2rgb.txt' />
        <param name="HSV_h1" value="$(arg HSV_h1)" />
        <param name="HSV_h2" value="$(arg HSV_h2)" />
        <param name="HSV_s1" value="$(arg HSV_s1)" />
        <param name="HSV_s2" value="$(arg HSV_s2)" />
        <param name="HSV_v1" value="$(arg HSV_v1)" />
        <param name="HSV_v2" value="$(arg HSV_v2)" />
        <param name='rgb_width' value='640' />
        <param name='rgb_height' value='480' />
        <param name='thermal_width' value='640' />
        <param name='thermal_height' value='512' />
        <param name='min_distance' value='1.0' />
        <param name='max_distance' value='300.0' />
        <param name='tp_in_rgb' value='/rgb_cam/image_raw' />
        <param name='tp_in_rgb_camerainfo' value='/rgb_cam/camera_info' />
        <param name='tp_in_thermal' value='/thermal_cam/thermal_image' />
        <param name='tp_in_thermal_camerainfo' value='/thermal_cam/camera_info' />
        <param name='tp_in_livox' value='/livox/lidar' />
        <param name='tp_out_rgb' value='/sensors_calib/image_rect' />
        <param name='tp_out_rgb_camerainfo' value='/sensors_calib/image_rect/camera_info' />
        <param name='tp_out_thermal' value='/sensors_calib/thermal_rect' />
        <param name='tp_out_thermal_camerainfo' value='/sensors_calib/thermal_rect/camera_info' />
        <param name='tp_out_depth_rgb' value='/sensors_calib/depth_rgb' />
        <param name='tp_out_depth_thermal' value='/sensors_calib/depth_thermal' />
        <param name='tp_out_intensity_rgb' value='/sensors_calib/intensity_rgb' />
        <param name='tp_out_intensity_thermal' value='/sensors_calib/intensity_thermal' />
    </node>
  </group>
</launch>

